pkg/promhttputil/merge.go

```diff
@@ -171,6 +171,93 @@ func MergeValues(antiAffinityBuffer model.Time, a, b model.Value, preferMax bool
 	return nil, fmt.Errorf("unknown type! %v", reflect.TypeOf(a))
 }
 
+// SumValues sums values `a` and `b` with the given antiAffinityBuffer
+func SumValues(antiAffinityBuffer model.Time, a, b model.Value, preferMax bool) (model.Value, error) {
```

Comment:
why is this needed as opposed to reusing promxy's constructs of combining subqueries when it issues it across multiple server groups?

Reply:
This only differs for instant query where results are merged for same vector from different tenant.
When promxy combine subqueries from different server groups, it actually take max in vector.